src/ImageSharp/Common/Helpers/ImageMaths.cs

```diff
@@ -39,14 +39,39 @@ public static int LeastCommonMultiple(int a, int b)
             return (a / GreatestCommonDivisor(a, b)) * b;
         }
 
+        /// <summary>
+        /// Calculates <paramref name="x"/> % 4
+        /// </summary>
+        [MethodImpl(InliningOptions.ShortMethod)]
+        public static int Modulo4(int x) => x & 3;
+
+        /// <summary>
+        /// Calculates <paramref name="x"/> % 8
+        /// </summary>
+        [MethodImpl(InliningOptions.ShortMethod)]
+        public static int Modulo8(int x) => x & 7;
+
+        /// <summary>
+        /// Fast (x mod m) calculator, with the restriction that
+        /// <paramref name="m"/> should be power of 2.
+        /// </summary>
+        [MethodImpl(InliningOptions.ShortMethod)]
+        public static int ModuloP2(int x, int m)
+        {
+            return x & (m - 1);
+        }
+
+        [MethodImpl(InliningOptions.ShortMethod)]
+        public static float Clamp(float x, float min, float max) => Math.Min(max, Math.Max(min, x));
```

Comment:
You might wanna benchmark this. The `IComparableExtensions` version should be a good bit faster.

I do, however want to ditch the extension method for more clear `MathUtils.Clamp***` methods.

Reply:
Good point, will do compare!

Reply:
Do we need the duplication?

Reply:
The one in `ComparableExtensions` is faster, I'm removing this!